rust/mlt-nom/src/v0x01/root.rs

```diff
@@ -0,0 +1,251 @@
+                    (input, value) = Stream::parse(input)?;
+                    properties.push(Property::raw(name, optional, RawPropValue::I64(value)));
+                }
+                ColumnType::U64 | ColumnType::OptU64 => {
+                    (input, optional) = parse_optional(column.typ, input)?;
+                    (input, value) = Stream::parse(input)?;
+                    properties.push(Property::raw(name, optional, RawPropValue::U64(value)));
+                }
+                ColumnType::F32 | ColumnType::OptF32 => {
+                    (input, optional) = parse_optional(column.typ, input)?;
+                    (input, value) = Stream::parse(input)?;
+                    properties.push(Property::raw(name, optional, RawPropValue::F32(value)));
+                }
+                ColumnType::F64 | ColumnType::OptF64 => {
+                    (input, optional) = parse_optional(column.typ, input)?;
+                    (input, value) = Stream::parse(input)?;
+                    properties.push(Property::raw(name, optional, RawPropValue::F64(value)));
+                }
+                ColumnType::Str | ColumnType::OptStr => {
+                    (input, stream_count) = utils::parse_varint::<usize>(input)?;
+                    (input, optional) = parse_optional(column.typ, input)?;
+                    // if optional has a value, one stream has already been consumed
+                    let stream_count = stream_count - usize::from(optional.is_some());
+                    let value_vec;
+                    (input, value_vec) = Stream::parse_multiple(input, stream_count)?;
+                    properties.push(Property::raw(name, optional, RawPropValue::Str(value_vec)));
+                }
+                ColumnType::Struct => {
+                    todo!("Struct column type not implemented yet");
+                }
+            }
+        }
+        if input.is_empty() {
+            Ok(RawFeatureTable {
+                name: layer_name,
+                extent,
+                id: id_stream.unwrap_or_default(),
+                geometry: geometry.ok_or(MltError::MissingGeometry)?,
+                properties,
+            })
+        } else {
+            Err(MltError::TrailingLayerData(input.len()))
+        }
+    }
+}
+
+fn parse_columns_meta(
+    mut input: &'_ [u8],
+    column_count: usize,
+) -> MltRefResult<'_, (Vec<Column<'_>>, usize)> {
+    #[allow(clippy::enum_glob_use)]
+    use ColumnType::*;
+
+    let mut col_info = Vec::with_capacity(column_count);
+    let mut geometries = 0;
+    let mut ids = 0;
+    for _ in 0..column_count {
+        let typ;
+        (input, typ) = Column::parse(input)?;
+        match typ.typ {
+            Geometry => geometries += 1,
+            Id | OptId | LongId | OptLongId => ids += 1,
+            _ => {}
+        }
+        col_info.push(typ);
+    }
+    if geometries > 1 {
+        return Err(MltError::MultipleGeometryColumn);
```

Comment:
nit: to be consistent

```suggestion
        return Err(MltError::MultipleGeometryColumns);
```

Reply:
thx